extensions/resource-deployment/src/ui/deployAzureSQLVMWizard/pages/azureSettingsPage.ts

```diff
@@ -0,0 +1,307 @@
+	}
+
+	public async initialize() {
+		this.pageObject.registerContent(async (view: azdata.ModelView) => {
+
+			await Promise.all([
+				this.createAzureAccountsDropdown(view),
+				this.createAzureSubscriptionsDropdown(view),
+				this.createResourceDropdown(view),
+				this.createAzureRegionsDropdown(view)
+			]);
+			this.populateAzureAccountsDropdown();
+
+			this._form = view.modelBuilder.formContainer()
+				.withFormItems(
+					[
+						{
+							component: this.wizard.createFormRowComponent(view, constants.AzureAccountDropdownLabel, '', this._azureAccountsLoader, true)
+						},
+						{
+							component: this.buttonFlexContainer
+						},
+						{
+							component: this.wizard.createFormRowComponent(view, constants.AzureAccountSubscriptionDropdownLabel, '', this._azureSubscriptionLoader, true)
+						},
+						{
+							component: this.wizard.createFormRowComponent(view, constants.AzureAccountResourceGroupDropdownLabel, '', this._resourceGroupDropdown, true)
+						},
+						{
+							component: this.wizard.createFormRowComponent(view, constants.AzureAccountRegionDropdownLabel, '', this._azureRegionsLoader, true)
+						}
+					],
+					{
+						horizontal: false,
+						componentWidth: '100%'
+					})
+				.withLayout({ width: '100%' })
+				.component();
+			return view.initializeModel(this._form);
+		});
+	}
+
+	public async onEnter(): Promise<void> {
+		this.wizard.wizardObject.registerNavigationValidator((pcInfo) => {
+			return true;
+		});
+	}
+
+	public onLeave(): void {
+		this.wizard.wizardObject.registerNavigationValidator((pcInfo) => {
+			return true;
+		});
+	}
+
+	private async createAzureAccountsDropdown(view: azdata.ModelView) {
+
+		this._azureAccountsDropdown = view.modelBuilder.dropDown().withProperties({}).component();
+
+		this._azureAccountsDropdown.onValueChanged(async (value) => {
+			this.wizard.model.azureAccount = this._accountsMap.get(value.selected)!;
+			this.populateAzureSubscriptionsDropdown();
+		});
+
+		this.signInButton = view.modelBuilder.button().withProperties<azdata.ButtonProperties>({
+			label: 'Sign In',
+			width: '100px'
+		}).component();
+		this.refreshButton = view.modelBuilder.button().withProperties<azdata.ButtonProperties>({
+			label: 'Refresh',
+			width: '100px'
+		}).component();
+
+		this.signInButton.onDidClick(async (event) => {
+			await vscode.commands.executeCommand('workbench.actions.modal.linkedAccount');
+			await this.populateAzureAccountsDropdown();
+		});
+		this.refreshButton.onDidClick(async (event) => {
+			await this.populateAzureAccountsDropdown();
+		});
+		this.buttonFlexContainer = view.modelBuilder.flexContainer().withLayout({
+			flexFlow: 'row'
+		}).withItems([this.signInButton, this.refreshButton], { CSSStyles: { 'margin-right': '5px', } }).component();
+
+		this._azureAccountsLoader = view.modelBuilder.loadingComponent().withItem(this._azureAccountsDropdown).component();
+	}
+
+	private async populateAzureAccountsDropdown() {
```

Comment:
would the method that exists in modelviewutlis.ts not work for your usage?

Reply:
I had to add custom onChange listener. For that reason, I decided to implement my own components.

Reply:
And why wouldn't you just modify that one to allow hooking into the onChange listener? Duplicating code like this is not a great habit to get in to and makes maintaining it much more annoying.